ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -232,9 +232,9 @@ private[kvutils] class TransactionCommitter(
             transactionEntry.submissionSeed,
           )
           .consume(
-            lookupContract(transactionEntry, commitContext.inputs),
-            lookupPackage(transactionEntry, commitContext.inputs),
-            lookupKey(commitContext.inputs, knownKeys),
+            lookupContract(transactionEntry, commitContext),
+            lookupPackage(transactionEntry, commitContext),
+            lookupKey(commitContext, knownKeys),
```

Comment:
I'd prefer either testing the property at this level (i.e. testing `validateModelConformance` for it rather than `CommitContext.read`) or encapsulating `inputs` in some way, so that `TransactionCommitter` changes cannot cause a regression either.

Reply:
I agree with hiding and encapsulating `inputs`. It's clear that direct access is dangerous.

Reply:
That's a great idea -- I'll encapsulate `inputs` so that we can be certain we record all read keys.